search-provider/eks-metadata-provider.c

```diff
@@ -406,41 +411,46 @@ translate_gvariant_to_gvalue_parse_enum (GVariant  *variant,
 static gboolean
 append_construction_prop_from_variant (const char                   *key,
                                        GVariant                     *variant,
-                                       GArray                       *values_array,
-                                       GPtrArray                    *props_array,
+                                       GArray                       *parameters_array,
                                        VariantToValueTransformFunc   transform,
                                        gpointer                      user_data,
                                        GError                      **error)
 {
-  guint index = values_array->len;
+  guint index = parameters_array->len;
+  GParameter *parameter = NULL;
   GValue *value = NULL;
 
-  /* Need to be careful to set value *after* the array has been resized */
-  g_array_set_size (values_array, values_array->len + 1);
-  value = &(g_array_index (values_array, GValue, index));
+  /* Need to be careful to set parameter *after* the array has been resized */
+  g_array_set_size (parameters_array, parameters_array->len + 1);
+  parameter = &g_array_index (parameters_array, GParameter, index);
+  value = &parameter->value;
 
   /* If we return false from here, this should be treated as an
    * unrecoverable error for the call */
   if (!transform (variant, value, user_data, error))
     return FALSE;
 
-  g_ptr_array_add (props_array, g_strdup (key));
+  /* Setting a const value without a copy here seems suspicious,
+   * but it should be fine as long as the passed in value outlives
+   * the parameters_array outparam.
+   *
+   * In this case, all the values that should be passed here live in
+   * static storage (see model_variant_types) or the .data */
+  parameter->name = key;
```

Comment:
It's a bit suspicious that in the other case we were copying the strings instead... The fact that the `GParameter` has a `const char *` field does not actually prevent you from assigning a new allocation of memory to it; as long as you know that it's the case you can just as well cast that pointer back to `char *` and free that memory after creating the object.
I'd actually lean towards being conservative here and copy the strings.

Reply:
Fair enough, I suppose I could define a clear func to deal with this.